subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -21,21 +21,27 @@ import org.gradle.api.artifacts.Dependency
 import org.gradle.api.artifacts.ExternalModuleDependency
 import org.gradle.api.artifacts.ModuleDependency
 import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.plugins.ExtensionAware
 
 
 /**
  * Receiver for `dependencies` block providing convenient utilities for configuring dependencies.
  *
  * @see [DependencyHandler]
  */
-class DependencyHandlerScope
+open class DependencyHandlerScope
 private constructor(
     val dependencies: DependencyHandler
 ) : DependencyHandler by dependencies {
 
     companion object {
         fun of(dependencies: DependencyHandler) =
-            DependencyHandlerScope(dependencies)
+            ExtensionAwareDependencyHandlerScope(dependencies) as DependencyHandlerScope
```

Comment:
Instead of casting (which then I think needs to happen at runtime as well (but I might be wrong)).
Just explicitly declare the return type.
```kotlin
fun of(dependencies: DependencyHandler): DependencyHandlerScope
```